curp/src/server/raw_curp/mod.rs

```diff
@@ -259,6 +262,14 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
             }
         });
 
+        // check if commit_index needs to be updated
+        if self.can_update_commit_index_to(&log_w, index, self.term()) && index > log_w.commit_index
+        {
+            log_w.commit_index = index;
+            debug!("{} updates commit index to {index}", self.id());
+            self.apply(&mut *log_w);
```

Comment:
Why should we apply log in `handle_propose`?

Reply:
special case for single node cluster